connection.go

```diff
@@ -873,3 +880,46 @@ func (c *Connection) closeNetwork() {
 func (c *Connection) getLastActivityTime() time.Time {
 	return time.Unix(0, c.lastActivity.Load())
 }
+
+func (c *Connection) sendBufSize() (sendBufUsage int, sendBufSize int, _ error) {
+	sendBufSize = -1
+	sendBufUsage = -1
+
+	if c.sysConn == nil {
+		return sendBufUsage, sendBufSize, errNoSyscallConn
+	}
+
+	var (
+		// current send buffer usage
+		ioctlErr error
+
+		// current send buffer limit
+		sockoptErr error
+	)
+
+	errs := c.sysConn.Control(func(fd uintptr) {
+		sendBufUsage, ioctlErr = unix.IoctlGetInt(int(fd), unix.SIOCOUTQ)
+		sendBufSize, sockoptErr = unix.GetsockoptInt(int(fd), unix.SOL_SOCKET, unix.SO_SNDBUF)
+	})
+
+	errs = multierr.Append(errs, sockoptErr)
+	errs = multierr.Append(errs, ioctlErr)
+	return sendBufUsage, sendBufSize, errs
+}
+
+func getSysConn(conn net.Conn, log Logger) syscall.RawConn {
+	connSyscall, ok := conn.(syscall.Conn)
+	if !ok {
+		log.WithFields(LogField{"connectionType", fmt.Sprintf("%T", conn)}).
+			Error("Connection does not implement SyscallConn.")
+		return nil
+	}
+
+	sysConn, err := connSyscall.SyscallConn()
```

Comment:
`*net.TCPConn` and others already have the SyscallConn method so the indirection via syscall.Conn could be avoided.

Reply:
We're not dealing with a `*net.TCPConn`, but a `net.Conn`, so we need to cast. TChannel allows both customization of the dialer which returns any `net.Conn`, so using `syscall.Conn` is more general than casting to a `*net.TCPConn`.

Reply:
I was actually thinking more along the lines of testing abstractly for presence of the `SyscallConn` method rather than concretely for `*net.TCPConn`. Having actually looked at the definition of `syscall.Conn` I realized the code as written already does that effectively (I expected the `syscall.Conn` interface to be larger).